### PR TITLE
Make days in StoryPeriod collapsible

### PR DESCRIPTION
**Pull request by simfeld**

### CODE REVIEW

**Comment by simfeld** [comment]
image

File: frontend/src/router.js [hunk 1]
```diff
@@ -245,34 +252,28 @@ export default new Router({
         {
           path: 'story',
           name: 'camp/story',
+          async beforeEnter(to, from, next) {
+            redirectToPeriod(to, from, next, 'camp/story/period')
+          },
+        },
+        {
+          path: 'story/period/:periodId/:periodTitle?',
```
Comment (simfeld) [inline frontend/src/router.js]: The period is now part of the route for the story overview

File: frontend/src/components/program/PeriodSwitcher.vue [hunk 2]
```diff
@@ -53,6 +53,7 @@ export default {
       type: Function,
       required: true,
     },
+    routeName: { type: String, default: 'camp/program/period' },
```
Comment (simfeld) [inline frontend/src/components/program/PeriodSwitcher.vue]: Make it reusable for story

File: frontend/src/components/story/StoryDay.vue [hunk 3]
```diff
@@ -1,54 +1,59 @@
...
-            />
-          </api-form>
-        </div>
+  <v-expansion-panel>
+    <v-expansion-panel-header>
+      <h3>{{ dateLong(day.start) }}</h3>
+    </v-expansion-panel-header>
+    <v-expansion-panel-content>
+      <template v-if="loading">
+        <v-skeleton-loader class="mt-2 mt-sm-3" type="list-item-three-line" />
```
Comment (carlobeltrame) [inline frontend/src/components/story/StoryDay.vue]: You could omit the `<template>` here and move the `v-if` to the `<v-skeleton-loader>`

File: frontend/src/components/story/StoryDay.vue [hunk 4]
```diff
@@ -93,10 +98,18 @@ export default {
...
+  async mounted() {
+    // refresh to get new schedule entries
+    await this.day.scheduleEntries().$reload()
+    // refresh individual schedule entries to get new story content nodes
+    await Promise.all(
+      this.day.scheduleEntries().items.map((entry) =>
+        entry
+          .activity()
+          .contentNodes()
+          ._meta.load.then((contentNodes) => this.api.reload(contentNodes))
```
Comment (carlobeltrame) [inline frontend/src/components/story/StoryDay.vue:112]: This is an n+1 query, but it was one before, so you don't have to fix that in this PR

File: frontend/src/components/story/StoryPeriod.vue [hunk 5]
```diff
@@ -22,5 +18,27 @@ export default {
...
+      handler(value) {
+        this.computeExpandedDays(value)
+      },
+    },
+  },
+  methods: {
+    computeExpandedDays(period) {
+      period.days()._meta.load.then((days) => {
+        this.expandedDays = days.items.map((day, idx) =>
+          Date.parse(day.end) >= new Date() ? idx : null
```
Comment (carlobeltrame) [inline frontend/src/components/story/StoryPeriod.vue]: Possible improvement: Once the period has finished completely, the days could all be expanded again

**Review by BacLuc — APPROVED** [review]

File: frontend/src/router.js [hunk 6]
```diff
@@ -228,9 +228,16 @@ export default new Router({
           component: () =>
             import(/* webpackChunkName: "campAdmin" */ './views/camp/Admin.vue'),
         },
+        {
+          path: 'program',
+          name: 'camp/program',
+          async beforeEnter(to, from, next) {
+            redirectToPeriod(to, from, next, 'camp/program/period')
+          },
+        },
```
Comment (carlobeltrame) [inline frontend/src/router.js]: The order of these routes matters. Did you have an explicit reason to move this up?

**Review by carlobeltrame — APPROVED** [review]

File: frontend/src/components/story/StoryDay.vue (hunk 3 above)
Comment (simfeld) [reply frontend/src/components/story/StoryDay.vue]: [8608020](https://github.com/ecamp/ecamp3/pull/3038/commits/86080202531df5b06be0648e5ebfc87aebdac007)

File: frontend/src/components/story/StoryPeriod.vue (hunk 5 above)
Comment (simfeld) [reply frontend/src/components/story/StoryPeriod.vue]: Good point => [f7edaf4](https://github.com/ecamp/ecamp3/pull/3038/commits/f7edaf4131cbd0ed1ba471f198a26da7ea36b681)

File: frontend/src/router.js (hunk 6 above)
Comment (simfeld) [reply frontend/src/router.js]: [db6bb6c](https://github.com/ecamp/ecamp3/pull/3038/commits/db6bb6ce2db90be0ce9a124f207ce9751d8dfc8c), I checked the routing behavior and it looks ok to me

File: frontend/src/router.js [hunk 7]
```diff
@@ -243,36 +250,30 @@ export default new Router({
           props: (route) => ({ camp: campFromRoute(route) }),
         },
         {
-          path: 'story',
-          name: 'camp/story',
+          path: 'story/period/:periodId/:periodTitle?',
+          name: 'camp/story/period',
```
Comment (manuelmeister) [inline frontend/src/router.js]:
```suggestion
          name: 'camp/period/story',
```
Wie auch bei den anderen Namen.

File: frontend/src/router.js [hunk 8]
```diff
@@ -243,36 +250,30 @@ export default new Router({
...
           component: () =>
             import(/* webpackChunkName: "campStory" */ './views/camp/Story.vue'),
+          beforeEnter: requirePeriod,
+        },
+        {
+          path: 'story',
+          name: 'camp/story',
```
Comment (manuelmeister) [inline frontend/src/router.js:235]: Das kannst du so lassen.

File: frontend/src/router.js [hunk 9]
```diff
@@ -243,36 +250,30 @@ export default new Router({
...
-              next(campRoute(camp, 'admin', to.query))
-            }
-          },
-        },
         {
           path: 'dashboard',
           name: 'camp/dashboard',
           component: () =>
-            import(/* webpackChungName: "camp" */ './views/camp/Dashboard.vue'),
+            import(/* webpackChunkName: "camp" */ './views/camp/Dashboard.vue'),
```
Comment (manuelmeister) [inline frontend/src/router.js]: remove webpack chunk names

**Review by manuelmeister — CHANGES_REQUESTED** [review]

File: frontend/src/router.js (hunk 7 above)
Comment (simfeld) [reply frontend/src/router.js]: [acad08d](https://github.com/ecamp/ecamp3/pull/3038/commits/acad08ddce9ea3f89ea2ac3b9fb2bdc9de318397)

File: frontend/src/router.js (hunk 9 above)
Comment (simfeld) [reply frontend/src/router.js]: [a972bcb](https://github.com/ecamp/ecamp3/pull/3038/commits/a972bcbbef07b8fc7d64dd83b5cd6ffc7c15f8ab)

**Review by carlobeltrame — APPROVED** [review]

**Review by manuelmeister — APPROVED** [review]

**Pull request merged**